examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,552 @@
+```
+### TEI Reranking Service
+
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-teirerank 8808:80
+```
+Follow the below steps in a different terminal.
+
+The TEI Reranking Service reranks the documents returned by the retrieval
+service. It consumes the query and list of documents and returns the document
+index based on decreasing order of the similarity score. The document
+corresponding to the returned index with the highest score is the most relevant
+document for the input query.
+```
+curl http://localhost:8808/rerank \
+    -X POST \
+    -d '{"query":"What is Deep Learning?", "texts": ["Deep Learning is not...", "Deep learning is..."]}' \
+    -H 'Content-Type: application/json'
+```
+
+Output is:  `[{"index":1,"score":0.9988041},{"index":0,"score":0.022948774}]`
+
+
+### TGI Service
+
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-tgi 9009:80
+```
+Follow the below steps in a different terminal.
+
+```
+curl http://localhost:9009/generate \
+  -X POST \
+  -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":17, "do_sample": true}}' \
+  -H 'Content-Type: application/json'
+```
+
+TGI service generates text for the input prompt. Here is the expected result from TGI:
+
+```
+{"generated_text":"We have all heard the buzzword, but our understanding of it is still growing. It’s a sub-field of Machine Learning, and it’s the cornerstone of today’s Machine Learning breakthroughs.\n\nDeep Learning makes machines act more like humans through their ability to generalize from very large"}
+```
+
+**NOTE**: After TGI service is started, it takes few minutes to load a LLM model and warm up, before reaching `Ready` state.
+
+If you get
+
+```
+curl: (7) Failed to connect to localhost port 8008 after 0 ms: Connection refused
+```
+
+and the log shows model warm up, please wait for a while and try it later.
+
+```
+2024-06-05T05:45:27.707509646Z 2024-06-05T05:45:27.707361Z  WARN text_generation_router: router/src/main.rs:357: `--revision` is not set
+2024-06-05T05:45:27.707539740Z 2024-06-05T05:45:27.707379Z  WARN text_generation_router: router/src/main.rs:358: We strongly advise to set it to a known supported commit.
+2024-06-05T05:45:27.852525522Z 2024-06-05T05:45:27.852437Z  INFO text_generation_router: router/src/main.rs:379: Serving revision bdd31cf498d13782cc7497cba5896996ce429f91 of model Intel/neural-chat-7b-v3-3
+2024-06-05T05:45:27.867833811Z 2024-06-05T05:45:27.867759Z  INFO text_generation_router: router/src/main.rs:221: Warming up model
+```
+
+### Dataprep Microservice (Advanced)
+
+Add Knowledge Base via HTTP Links:
+
```

Comment:
Please add 

Once you have set up port forward for the dataprep service, you can upload, delete, and list documents.